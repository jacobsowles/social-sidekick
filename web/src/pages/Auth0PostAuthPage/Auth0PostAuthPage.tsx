import { Auth0Error, Auth0UserProfile } from 'auth0-js';
import React, { Dispatch, PureComponent } from 'react';
import { InjectedAlertProp, withAlert } from 'react-alert';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import { Action } from 'redux';

import { setUserState } from '@actions/user.actions';
import AuthService from '@core/auth';

interface Auth0PostAuthPageOwnProps {
  alert: InjectedAlertProp;
}

interface Auth0PostAuthPageDispatchProps {
  getUser: () => void;
}

type Auth0PostAuthPageProps = Auth0PostAuthPageOwnProps & Auth0PostAuthPageDispatchProps;

class Auth0PostAuthPage extends PureComponent<Auth0PostAuthPageProps> {
  public componentDidMount() {
    this.props.getUser();
  }

  public render() {
    return <Redirect to="/dashboard" />;
  }
}

const mapDispatchToProps = (dispatch: Dispatch<Action>, ownProps: Auth0PostAuthPageOwnProps) => {
  return {
    getUser: () => {
      const authService = new AuthService();
      authService.getUser((error: Auth0Error, user: Auth0UserProfile) => {
        if (user) {
          dispatch(setUserState(user));
        } else {
          ownProps.alert.error(error.error);
        }
      });
    }
  };
};

export default withAlert(
  connect(
    null,
    mapDispatchToProps
  )(Auth0PostAuthPage)
);
