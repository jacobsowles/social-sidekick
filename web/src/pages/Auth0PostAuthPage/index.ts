import Auth0PostAuthPage from './Auth0PostAuthPage';
export default Auth0PostAuthPage;
