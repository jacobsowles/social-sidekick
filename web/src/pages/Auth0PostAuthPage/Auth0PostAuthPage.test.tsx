import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import { mockStore, init, shallow, toJson } from '@tests/component-test-base';
import Auth0PostAuthPage from '@pages/Auth0PostAuthPage';

describe('Auth0PostAuthPage', () => {
  init(
    () => (
      <Provider store={mockStore({})}>
        <BrowserRouter>
          <Auth0PostAuthPage />
        </BrowserRouter>
      </Provider>
    ),
    'div'
  );

  it('should render the component without crashing', () => {
    toJson(shallow()).should.matchSnapshot();
  });
});
