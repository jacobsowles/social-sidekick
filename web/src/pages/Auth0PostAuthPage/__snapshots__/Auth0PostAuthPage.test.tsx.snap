// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth0PostAuthPage should render the component without crashing 1`] = `
<BrowserRouter>
  <WithAlert(Connect(Auth0PostAuthPage)) />
</BrowserRouter>
`;
